pyscraper/new_hansard.py

```diff
@@ -0,0 +1,1442 @@
+            member_tags = tag.xpath('.//ns:Member', namespaces=self.ns_map)
+            if len(member_tags) == 1:
+                member_tag = member_tags[0]
+        elif tag_name == 'Member':
+            member_tag = tag
+
+        if member_tag is not None:
+            if member_tag.get('PimsId') == '-1':
+                return self.handle_minus_member(member_tag)
+            if member_tag.get('PimsId') == '0':
+                return None
+            member = self.resolver.match_by_pims(member_tag.get('PimsId'))
+            if member is not None:
+                member['person_id'] = member.get('id')
+                member['name'] = self.resolver.name_on_date(member['person_id'], self.date)
+                return member
+            else:
+                raise ContextException(
+                    'No match for PimsId {0}\n'.format(member_tag.get('PimsId')),
+                    stamp=tag.get('url'),
+                    fragment=member_tag.text
+                )
+
+        return None
+
+    def parse_date(self, date):
+        text = self.get_single_line_text_from_element(date)
+        time_parts = re.match('\s*the\s+house (?:being |having )?met at?\s+(.*?)$(?i)', text)
+        if time_parts:
+            time = time_parts.group(1)
+            time = re.sub('</?i>',' ', time)
+            time = re.sub('\s+',' ', time)
+            if re.match("half-past Nine(?i)", time):
+                    newtime = '09:30:00'
+            elif re.match("a quarter to Ten o(?i)", time):
+                    newtime = '09:45:00'
+            elif re.match("Ten o'clock(?i)", time):
+                    newtime = '10:00:00'
+            elif re.match("half-past Ten(?i)", time):
+                    newtime = '10:30:00'
+            elif re.match("Eleven o&#039;clock(?i)", time):
+                    newtime = '11:00:00'
+            elif re.match("twenty-five minutes past\s*Eleven(?i)", time):
+                    newtime = '11:25:00'
+            elif re.match("twenty-six minutes past\s*Eleven(?i)", time):
+                    newtime = '11:26:00'
+            elif re.match("twenty-nine minutes past\s*Eleven(?i)", time):
+                    newtime = '11:29:00'
+            elif re.match("half-past Eleven(?i)", time):
+                    newtime = '11:30:00'
+            elif re.match("Twelve noon(?i)", time):
+                    newtime = '12:00:00'
+            elif re.match("half-past One(?i)", time):
+                    newtime = '13:30:00'
+            elif re.match("half-past Two(?i)", time):
+                    newtime = '14:30:00'
+            elif re.match("twenty minutes to Three(?i)", time):
+                    newtime = '14:40:00'
+            elif re.match("10 minutes past Three(?i)", time):
+                    newtime = '15:10:00'
+            elif re.match("Six o'clock(?i)", time):
+                    newtime = '18:00:00'
+
+            self.time = newtime
+        elif re.match('.*was asked.*', text):
```

Comment:
This is presumably where they've used the date element to contain actual data? :-/